docs/dev/data_validation.rst

```diff
@@ -0,0 +1,688 @@
+of tables are changing simultaneously, it is helpful to be able to run the full ETL from
+scratch, run all of the data validation tests against the outputs, and use the results
+to update the test parameters (especially expected row counts) appropriately. This can
+be done by manually kicking off a PUDL deployment on your branch. To initiate a branch
+build, in the PUDL repo on GitHub go to `Actions
+<https://github.com/catalyst-cooperative/pudl/actions>`_ and select `build-deploy-pudl
+<https://github.com/catalyst-cooperative/pudl/actions/workflows/build-deploy-pudl.yml>`_.
+On the right hand side select Run Workflow and then select your branch in the dropdown
+and click the Run Workflow button. Shortly thereafter you should see a notification in
+the ``pudl-deployments`` channel in our Slack saying that the build has kicked off. It
+should take about 3 hours to complete. You can track its progress and watch the logs in
+the `Google Cloud Console
+<https://console.cloud.google.com/monitoring/dashboards/builder/992bbe3f-17e6-49c4-a9e8-8f1925d4ec24>`_.
+
+Getting fresh row counts from a branch build
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To catch unexpected changes to the data, we keep track of the expected number of rows in
+each data table we distribute. These expectations are stored in
+``dbt/seeds/etl_full_row_counts.csv`` and they can be updated using the ``dbt_helper``
+script based on the observed row counts in your local PUDL Parquet outputs. If you can't
+run the full ETL locally, the nightly builds / branch build also generate updated row
+count expectations. After a branch build completes, you can download the updated
+``etl_full_row_counts.csv`` file from the build outputs that are uploaded to
+``gs://builds.catalyst.coop/<build-id>/etl_full_row_counts.csv`` See the
+:doc:`nightly_data_builds` documentation for more details on accessing the nightly build
+outputs. Replace the ``etl_full_row_counts.csv`` in your local PUDL git repo with the
+one you've downloaded and use ``git diff`` to see what has changed. Make sure to review
+the row count changes closely to see if there's anything unexpected.
+
+--------------------------------------------------------------------------------
+Debugging data validation failures
+--------------------------------------------------------------------------------
+
+* Using output from ``dbt_helper validate``.
+* By inspecting and running the compiled SQL yourself.
+* Explain What "compiled" SQL means here.
+* Using ``--store-failures`` and the ``pudl_dbt_tests.duckdb`` output -- what is
+  stored in that database anyway?
+* Using ``duckdb < path/to/compiled.sql``
+* Using DuckDB's ``.read path/to/compiled.sql`` to play with data interactively.
+* Go through a simpler example before getting into the complicated quantile checks test.
+
+Debugging quantile checks
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. todo::
+
+  This seems quite involved. Can we make it simpler? Improve the test failure output to
```

Comment:
This section probably went out of date with https://github.com/catalyst-cooperative/pudl/pull/4386 and should be updated based on that new functionality.